dev-util/gn/gn-01463.recipe

```diff
@@ -0,0 +1,62 @@
+SUMMARY="A meta-build system that generates build files for Ninja"
+DESCRIPTION="GN is a meta-build system that generates build files for Ninja"
+HOMEPAGE="https://gn.googlesource.com/"
+COPYRIGHT="2008-2018 The Chromium Authors."
+LICENSE="BSD (3-clause)"
+REVISION="1"
+srcGitRev="091169beda92aff21ded3e9dfb392ea12b10c0e0"
+SOURCE_URI="https://github.com/return/gn/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="e17c58a90e26212c74cd527b1a10d315ede15a64c9cf0755b4ef78418c379b76"
+SOURCE_FILENAME="gn-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="gn-$srcGitRev"
+PATCHES="gn-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+SECONDARY_ARCHITECTURES="x86"
+fi
+
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	gn$secondaryArchSuffix = $portVersion
+	cmd:gn = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:g++$secondaryArchSuffix
+	cmd:ninja
+	cmd:python
+	"
+
+BUILD()
+{
+	export CC=gcc
+	export CXX=g++
+
+	python build/gen.py --host haiku --platform=haiku --no-sysroot
+	ninja -C out $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $commandBinDir
+	cp out/gn $commandBinDir
+}
+
+TEST()
+{
+	echo "Test"
+	# TODO: GN unittests fail to run.
+	#./out/gn_unittests
```

Comment:
I'd be great if we can have an issue opened with the failure log from the testsuite. `TODO`s tend to be ignored...